### DISCUSSION TITLE
Expose Dropdown fields as enumerations via graphql

### DISCUSSION THREAD

**Discussion by shkreios** [directus/directus#5885] [Feature Requests]
# Feature
Currently, the graphql api exposes a dropdown fields as `String!`. It would be nice to have it instead exposed the type as an enumeration of all possible choices. If `Allow other values` is selected it could be set to a union over all choice & `String!`

# Usecase
We are using the strictly typed library [i18next](https://github.com/i18next/i18next) with typescript which allows us to have type checking on our i18n keys. For example

```ts
t('common:roles.technical-lead')
```

The role name here is dynamic and comes from directus i.e.
```ts
t(`common:roles.${member.role}` as const)
```

Currently, the type of the i18n key would be `common:roles.${string}`.

With role being an enumeration i.e `'managing-partner' | 'project-lead' | 'technical-lead'` it could be 
```ts
"common:roles.managing-partner" | "common:roles.project-lead" | "common:roles.technical-lead"
```

Thank you in advance.

**Comment by n-hallberg** [comment]
This would be a neat feature! We got confused by that too.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
